docs/prometheus-integration.md

```diff
@@ -0,0 +1,103 @@
+# Prometheus Integration
+
+## Purpose
+Prometheus server can monitor various metrics and provide an observation of the 
+Antrea Controller and Agent components. The doc provides general guidelines to 
+the configuration of Prometheus server to operate with the Antrea components.
+
+## About Prometheus
+[Prometheus](https://prometheus.io/) is an open source monitoring and alerting 
+server. Prometheus is capable of collecting metrics from various Kubernetes 
+components, storing and providing alerts.
+Prometheus can provide visibility with integration to other products such as 
+[Grafana](https://grafana.com/).
+ 
+One of Prometheus capabilities is self-discovery of Kubernetes services which
+expose their metrics. So Prometheus can 
+scrape the metrics of any additional components which are added to the cluster
+without further configuration changes. 
+ 
+## Antrea Configuration
+Enable Prometheus metrics listener by setting `enablePrometheusMetrics` 
+parameter to true in the Controller and the Agent configurations.
+ 
```

Comment:
Does it make sense to have CLI showing the output of metrics http handler corresponding to agent/controller?

Reply:
@srikartati is this related to this PR? you may want to create a new issue if you have some improvement ideas for Prometheus integration

Reply:
Yes. Antrea agent and controller metrics are added to apiserver metrics http handle. I did not mean adding new CLI. Just show the CLI output of metrics handle using curl.

Reply:
I think that curling Prometheus client (on the agent or controller) or server is quite standard.
There's nothing specific to Antrea here IMO.
Are we doing anything unique which worths documenting?

Reply:
I'm fine punting that discussion to a future PR. The issue with adding sample outputs is that you may need to update them fairly often early on, when you are still hammering out details.

Reply:
We will be showing newly added antrea agent and controller prometheus metrics. Also, this will validate for the user that prometheus exporter is sending antrea metrics.

Reply:
@ksamoray up to you. maybe adding a sample `curl` command that can be used to query metrics and confirm everything is working is not a bad idea.